File: task_equivalents/llm_translations/gemini/optimized/python/statistical_modelling_analysis/01_linear_regression_inference.py
import pandas as pd
import statsmodels.formula.api as smf

auto = pd.read_csv('data/csv/Auto.csv', na_values='?')
auto = auto.dropna()

model1 = smf.ols('mpg ~ horsepower', data=auto).fit()
print(model1.summary())

new_data = pd.DataFrame({'horsepower': [98]})
pred = model1.get_prediction(new_data).summary_frame()

# Confidence interval
print(pred[['mean', 'mean_ci_lower', 'mean_ci_upper']])

# Prediction interval
print(pred[['mean', 'obs_ci_lower', 'obs_ci_upper']])

File: task_equivalents/llm_translations/gemini/optimized/r/statistical_modelling_analysis/01_linear_regression_inference.R
Auto <- read.csv('data/csv/Auto.csv', na.strings = '?')
Auto <- na.omit(Auto)

model1_fit <- lm(mpg ~ horsepower, data = Auto)
print(summary(model1_fit))

xnew <- data.frame(horsepower = 98)

pred_ci <- predict(model1_fit, newdata = xnew, interval = "confidence", level = 0.95)
print(pred_ci)

pred_pi <- predict(model1_fit, newdata = xnew, interval = "prediction", level = 0.95)
print(pred_pi)